tests/suites/test_suite_psa_crypto.function

```diff
@@ -6375,6 +6375,65 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE */
+void derive_key_type( int alg_arg,
```

Comment:
I'm not sure why this function's name ends with `_type`. It seems to me that what differentiates it from `derive_key` is the presence of the `expected_export` argument, to it seems to me that `derive_key_result` would probably be a better name.

Reply:
To be honest I don't remember why test function name ends with `_type`. Will change as suggested.

Reply:
While looking for the comment about generating test vectors I noticed that function code was imported also from https://github.com/ARMmbed/mbedtls/pull/5218.

Reply:
Ah OK, so let's keep it unchanged here, and discuss possible changes in #5218 instead then.

Reply:
This function is about deriving keys of different types, so the name makes sense to me.

Reply:
According to the comment in https://github.com/ARMmbed/mbedtls/pull/5139/commits/cb3eacd6913a721f348f73247fffbfa77ce2dd01 test function was taken from https://github.com/ARMmbed/mbedtls/pull/5218.